apps/survey/src/components/prompts/standard/SubmitPrompt.vue

```diff
@@ -13,31 +41,193 @@
         <v-icon class="pb-1">$add</v-icon>
       </v-btn>
       <v-divider vertical></v-divider>
-      <next-mobile :disabled="!isValid" @click="action('next')">
+      <next-mobile :disabled="isValidMobile" @click="action('next')">
         {{ $t('recall.actions.nav.submit') }}
       </next-mobile>
     </template>
   </card-layout>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue';
+import type { PropType } from 'vue';
+import { computed, defineComponent, ref, watch } from 'vue';
 
+import type { Prompt, Prompts } from '@intake24/common/prompts';
+import type { MealState } from '@intake24/common/types';
+import {
+  ReviewMealList,
+  ReviewMealListMobile,
+  ReviewMealListMobileScroll,
+  ReviewMealListScroll,
+} from '@intake24/survey/components/layouts';
 import { usePromptUtils } from '@intake24/survey/composables';
 
 import createBasePrompt from '../createBasePrompt';
 
 export default defineComponent({
   name: 'SubmitPrompt',
 
+  components: {
+    ReviewMealList,
+    ReviewMealListMobile,
+    ReviewMealListScroll,
+    ReviewMealListMobileScroll,
+  },
+
   mixins: [createBasePrompt<'submit-prompt'>()],
 
+  props: {
+    meals: {
+      type: Array as PropType<MealState[]>,
+      required: true,
+    },
+    prompt: {
+      type: Object as PropType<Prompt>,
+      required: true,
+    },
+    review: {
+      type: Object as PropType<Prompts['submit-prompt']['review']>,
+    },
+  },
+
   setup(props, ctx) {
     const { action } = usePromptUtils(props, ctx);
 
-    const isValid = true;
+    const reviewed = ref<string[]>([]);
+
+    const reviewedMobile = ref<string[]>([]);
+
+    const isValid = ref<boolean>(false);
+
+    const isValidMobile = ref<boolean>(false);
+
+    const reachedBottom = ref<boolean>(false);
+
+    const reachedBottomMobile = ref<boolean>(false);
+
+    const showReviewMealListCheckbox = computed(() => {
+      if (
+        props.prompt.component === 'submit-prompt' &&
```

Comment:
`props.prompt.component === 'submit-prompt'` this check won't be needed, if you remove props.prompt property, it will be typed as submit-prompt. Same for few more checks below, which can then be simplified